### PR TITLE
fixed issue 5834, autocomplete is not being decorated

### PR DESCRIPTION
**Pull request by kossel**
First of all, thanks for your contribution! :-)

Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you propose PR to correct branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

The cause is because the `.has-error` class is being overwrote by 2 other css rules:
* ant-select style
`.@{select-prefix-cls}-search__field`
* auto-complete style
```
.@{input-prefix-cls} {
 input();
}
```
the `input()` creates duplicated css attributes, because auto-complete component already inject a <Input /> which already has `.ant-input` class (exact same attributes)

Also updated the sample file so you can test

### CODE REVIEW

File: components/select/style/index.less [hunk 1]
```diff
@@ -248,7 +248,7 @@
     }
 
     .@{select-prefix-cls}-search__field {
-      border: 0;
+      border: 0 @border-style-base @border-color-base;
```
Comment (benjycui) [inline components/select/style/index.less]: Actually, this line of code is confusing.

Comment (kossel) [inline components/select/style/index.less]: hmm actually it should be just `border-width: 0;` because `border:0` equals to `border: 0px initial initial` which render a black rectangle and overwrites the default gray color.

should I do another PR for this small change?

Comment (benjycui) [inline components/select/style/index.less]: Yep, thanks.

Comment (kossel) [inline components/select/style/index.less]: https://github.com/ant-design/ant-design/pull/6030
